omod/src/test/java/org/openmrs/web/rest/controller/TagRestControllerTest.java

```diff
@@ -0,0 +1,102 @@
+package org.openmrs.web.rest.controller;
+
+import java.util.List;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Tag;
+import org.openmrs.api.TagService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.test.Util;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.openmrs.web.rest.TagRestTestConstants;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+public class TagRestControllerTest extends MainResourceControllerTest {
+	
+	TagService tagService;
+	
+	public static final String TAG_RESOURCE_DATASET = "TagResourceDataSet.xml";
+	
+	@Before
+	public void setup() throws Exception {
+		tagService = Context.getService(TagService.class);
+		executeDataSet(TAG_RESOURCE_DATASET);
+	}
+	
+	@Override
+	public String getURI() {
+		return TagRestTestConstants.TAG_URI;
+	}
+	
+	@Override
+	public String getUuid() {
+		return TagRestTestConstants.TAG_UUID;
+	}
+	
+	@Override
+	public long getAllCount() {
```

Comment:
this method needs to be implemented in the test, hence returning zero. It will actually throw an error, which is being caught below @wluyima